### PR TITLE
DCD-1531: JSM end-to-end test

### PR DESCRIPTION
**Pull request by yzha645**
## Pull request description

This PR adds end-to-end test for JSM.

## Checklist
- [ ] I have successful end to end tests run (with & without domain)
- [ ] I have added unit tests (if applicable)
- [ ] I have user documentation (if applicable)

### CODE REVIEW

**Comment by yzha645** [comment]
e2e successful runs: 
jsm: https://github.com/atlassian-labs/data-center-terraform/actions/runs/2389140513 
domain: https://github.com/atlassian-labs/data-center-terraform/actions/runs/2389291355
no domain: https://github.com/atlassian-labs/data-center-terraform/actions/runs/2390459985

File: .github/workflows/e2e-test-jsm.yaml [hunk 1]
```diff
@@ -0,0 +1,58 @@
+name: E2E Testing JSM
```
Comment (nghazali) [inline .github/workflows/e2e-test-jsm.yaml:1]: We already have two e2e tests (with/without domain). E2e tests are heavy and expensive. I think we should avoid adding more workload as much as possible. 

Maybe we can run one of tests, say E2e test without domain, using JSM. In this case we have involved JSM in integration test and in the same time did not add a new workload to our verification process and also Jira software is tested in E2e with domain test.

**Review by nghazali — CHANGES_REQUESTED** [review]

File: .github/workflows/e2e-test-jsm.yaml (hunk 1 above)
Comment (yzha645) [reply .github/workflows/e2e-test-jsm.yaml]: yup, make sense. Moved jsm to the domain one.

File: test/e2etest/installer_test.go [hunk 2]
```diff
@@ -21,7 +21,7 @@ func TestInstaller(t *testing.T) {
 
 	runInstallScript(testConfig.ConfigPath)
 
-	clusterHealthTests(t, testConfig)
+	clusterHealthTests(t, testConfig, 2)
```
[the remark this answers is not in the document]
Comment (nghazali) [reply test/e2etest/installer_test.go:24]: After another look, I think this refactor for removing the constant and passing as a parameter, adds no value to the code but creates more complexity considering we never change this value here. I think we should back to the original implementation for this part, leave the value as constant and use it in line #22 in `clusterHealthTests` method. 
If we need to make this value (`expectedNumberOfNodes`) variable then we can do this refactor at the time of need.

File: test/e2etest/test-config.tfvars.tmpl [hunk 3]
```diff
@@ -14,7 +14,7 @@ region = "{{.region}}"
 # (optional) List of the products to be installed.
 # Supported products are jira, confluence, bitbucket, and bamboo.
 # e.g.: products = ["jira", "confluence"]
-products = ["jira", "confluence", "bamboo", "bitbucket"]
+products = [{{.products}}]
```
Comment (nghazali) [inline test/e2etest/test-config.tfvars.tmpl:17]: good catch :)

**Review by nghazali — CHANGES_REQUESTED** [review]

File: test/e2etest/installer_test.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (yzha645) [reply test/e2etest/installer_test.go:24]: In theory, each cluster would have different number of nodes. In our case, the number happens to be the same for different test cases. But to keep it simple, I'm happy to revert back the change.

File: test/e2etest/test-config.tfvars.tmpl [hunk 4]
```diff
@@ -71,6 +71,9 @@ bamboo_dataset_url = "https://bamboo-test-datasets.s3.amazonaws.com/testing_data
 # Helm chart version of Jira
 #
 
+{{if .jsm}}jira_image_repository = "atlassian/jira-servicemanagement"
+jira_version_tag = "4.20"{{end}}
```
[the remark this answers is not in the document]
Comment (yzha645) [reply test/e2etest/test-config.tfvars.tmpl:75]: We don't have default version tag for JSM in Terraform module, and we don't have Helm chart for vanilla JSM either that we can lean on. 

See our documentation for JSM configuration: https://atlassian-labs.github.io/data-center-terraform/userguide/configuration/JIRA_CONFIGURATION/
image

**Review by nghazali — APPROVED** [review]

**Pull request merged**